File: 414. Third Maximum Number/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var thirdMax = function(nums) {
  nums = Array.from(new Set(nums));
  const len = nums.length;
  if (len <= 2) return Math.max(...nums);
  if (len === 3) return Math.min(...nums);
  nums = nums.sort((a, b) => b - a);
  return nums[2];
};

File: 414. Third Maximum Number/solution.go
import "sort"

func thirdMax(nums []int) int {
	m := make(map[int]bool)
	sort.Ints(nums)

	for j := len(nums) - 1; j >= 0; j-- {
		_, ok := m[nums[j]]
		if !ok {
			m[nums[j]] = true
			if len(m) >= 3 {
				return nums[j]
			}
		}
	}
	return nums[len(nums)-1]
}